core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -371,16 +371,23 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
             delete positions[msg.sender];
         }
 
-        // Transfer tokens & collateral and burn the redeemed tokens.
-        collateralCurrency.safeTransfer(msg.sender, totalRedeemableCollateral.rawValue);
-        tokenCurrency.safeTransferFrom(msg.sender, address(this), tokensToRedeem.rawValue);
-        tokenCurrency.burn(tokensToRedeem.rawValue);
+        // Take the min of the remaining collateral and the collateral "owed". If the contract is undercapitalized,
+        // the caller will get as much collateral as the contract can pay out.
+        FixedPoint.Unsigned memory payout = FixedPoint.min(
```

Comment:
+1 This is a great way to deal with this! even in the worst case the user can still withdraw. Coupling this with the rounding stuff would be ideal 👍